From cf17d6251653f4a98e7c4f904ea2f0bc0ecedd5c Mon Sep 17 00:00:00 2001
From: Duncan Hopkins <duncan@duncanhopkins.me.uk>
Date: Thu, 15 Oct 2020 12:14:57 +0100
Subject: [PATCH] meson: Add xcb-fixes to loader when using x11 and dri3. Fixes
 undefined symbol for xcb_xfixes_create_region in loader_dri3_helper.c

loader_dr3_helper.c uses xcb_xfixes_create_region() that requires dep_xcb_xfixes to link. This is dependent on with_platform_x11 and with_dri3.
But the source meson file does not set this up dependent on with_dri3.
The build was initialsed using platforms=x11 and gallium-drivers=zink,swrast.

Reviewed-by: Eric Anholt <eric@anholt.net>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/7164>

Upstream-Status: Backport [cf17d6251653f4a98e7c4f904ea2f0bc0ecedd5c]

---
 meson.build | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index cfe02fa6373..3cb3c904927 100644
--- a/meson.build
+++ b/meson.build
@@ -1782,7 +1782,8 @@ if with_platform_x11
       dep_xxf86vm = dependency('xxf86vm')
     endif
   endif
-  if (with_egl or (
+  if (with_egl or 
+      with_dri3 or (
       with_gallium_vdpau or with_gallium_xvmc or with_gallium_xa or
       with_gallium_omx != 'disabled'))
     dep_xcb_xfixes = dependency('xcb-xfixes')
-- 
2.17.1

